### PR TITLE
feat(splunk): Add tags and project to payload

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -35,5 +35,6 @@
     // tests is often in your way and misclicked
     "python.unitTest.pyTestEnabled": false,
     "python.unitTest.unittestEnabled": false,
-    "python.unitTest.nosetestsEnabled": false
+    "python.unitTest.nosetestsEnabled": false,
+    "restructuredtext.confPath": ""
```
Comment (dcramer) [inline .vscode/settings.json]: idk what this is but vscode just added it so... *shrugs*

File: src/sentry_plugins/splunk/plugin.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 from __future__ import absolute_import
 
-from sentry import http
+from sentry import http, tagstore
```
Comment (dcramer) [inline src/sentry_plugins/splunk/plugin.py]: only reasonably formal API I could find for this

**Comment by dcramer** [comment]
aside this was requested by a user, and i confirmed we can actually ship nested data types (which wasn't obvious in the docs before)

**Comment by dcramer** [comment]
screen shot 2018-12-10 at 5 03 16 pm

File: README.rst [hunk 3]
```diff
@@ -83,6 +83,16 @@ Slack
...
+Splunk
+------
+
+These docs are specific to a Splunk Cloud (self-service) instance. For more details on the HEC endpoint, see the `official docs <https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector#Send_data_to_HTTP_Event_Collector_on_Splunk_Cloud_instances>`_.
+
+- In your Splunk instance, navigate to Settings -> Data Inputs -> HTTP Event Collector
+- Create a new token; enter a name ("Sentry") and select an index ("main")
+- Under Global Settings, Enable the HEC endpoint noting the port (8088), and SSL requirement (true)
+- In Sentry, enable the Splunk plugin, and paste the newly acquired token.
+- For the endpoint, use https://input-[splunk-instance].cloud.splunk.com:8088
```
Comment (ctaintor) [inline README.rst]:
```suggestion
- For the endpoint, use https://input-[splunk-instance].cloud.splunk.com:8088
- Make sure that `enable indexer acknowledgement` is not checked.
```

Comment (dcramer) [reply README.rst]: Thanks!

(I know nothing about Splunk)

**Comment by dcramer** [comment]
@markstory i did a full QA - you can see the attached screenshot of the result in Splunk's UI

**Comment by markstory** [comment]
Does splunk make it easy to search tags as a list of tuples or would it be better if the tags were a dictionary?

**Comment by dcramer** [comment]
@markstory the backend treats tags as a list of tuples, so making it a dictionary will create situations where tags end up missing from the results

**Comment by dcramer** [comment]
Build passed - no idea why Travis/etc is broken, or even how to re-register the check, so im just merging.

**Pull request merged**